qualification/tied_array_channelised_voltage/test_weights.py

```diff
@@ -0,0 +1,99 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+"""Weights test."""
+
+import numpy as np
+import pytest
+from pytest_check import check
+
+from .. import CBFRemoteControl, TiedArrayChannelisedVoltageReceiver
+from ..reporter import Reporter
+
+
+@pytest.mark.requirements("CBF-REQ-0123")
+async def test_weight_mapping(
+    cbf: CBFRemoteControl,
+    receive_tied_array_channelised_voltage: TiedArrayChannelisedVoltageReceiver,
+    pdf_report: Reporter,
+) -> None:
+    """Test that each weight coefficient applies to the correct input.
+
+    Verification method
+    -------------------
+    Verification by means of test. Configure the dsim with a tone in a random
+    channel for each polarisation. For each input:
+
+    - Set the ``?gain`` to 1 for that input and to zero for all other inputs.
+    - Set one beam to use only that input, and all other beams to sum the remaining
+      inputs.
+    - Check that the tone is found in the correct channel of the chosen beam.
+    - Check that the other beams all output zero.
+    """
+    receiver = receive_tied_array_channelised_voltage
+    client = cbf.product_controller_client
+
+    rng = np.random.default_rng()
+    channels = rng.integers(1, receiver.n_chans, size=2)
+    channel_width = receiver.bandwidth / receiver.n_chans
+    freqs = receiver.center_freq + (channels - receiver.n_chans / 2) * channel_width
+    amplitude = 0.5  # Will probably saturate the PFB, but we don't care
+    signals = f"cw({amplitude}, {freqs[0]}); cw({amplitude}, {freqs[1]});"
+
+    pdf_report.step("Configure dsim with tones.")
+    await cbf.dsim_clients[0].request("signals", signals)
+    pdf_report.detail(f"Set dsim signals to {signals!r}.")
+
+    rng = np.random.default_rng()
+    for input_idx in range(receiver.n_inputs):
+        input_label = receiver.input_labels[input_idx]
+        channel = channels[input_idx % 2]
+        pdf_report.step(f"Testing input {input_idx} ({input_label}).")
+        await client.request("gain-all", "antenna-channelised-voltage", 0.0)
+        await client.request("gain", "antenna-channelised-voltage", input_label, 1.0)
+        pdf_report.detail(f"Set gain on {input_label} to 1.0, all others to 0.0.")
+        candidate_beams = [i for i, source_indices in enumerate(receiver.source_indices) if input_idx in source_indices]
+        assert candidate_beams, "No beam includes this input"
+        test_beam = rng.choice(candidate_beams)
+        test_beam_name = receiver.stream_names[test_beam]
+        pdf_report.detail(f"Using beam {test_beam_name} as the test beam.")
+
+        source_indices = receiver.source_indices[test_beam]
+        input_pos = source_indices.index(input_idx)
+        weights = [0.0] * len(source_indices)
+        weights[input_pos] = 1.0
+        inv_weights = [1.0 - w for w in weights]
```

Comment:
Must admit, this is neat. Although, why this approach specifically? 
* Are you accommodating for testing weights other than `1.0` in the future?
* And/or, why didn't you just do it in the for-loop below with an `else` case?

Reply:
> Are you accommodating for testing weights other than 1.0 in the future?

That's what NGC-1257 is for.

> And/or, why didn't you just do it in the for-loop below with an else case?

Either way works; but I figured then the special one will get lost in amongst all the rest, whereas with this approach it's clearer when reading the report how the one beam is being treated specially.